src/ElasticSearch/EloquentHitsIteratorAggregate.php

```diff
@@ -77,7 +78,22 @@ public function getIterator(): Traversable
 
                 $key = $source['__class_name'].'::'.$hit['_id'];
 
-                return isset($models[$key]) ? $models[$key] : null;
+                if (! isset($models[$key])) {
+                    return null;
+                }
+
+                /** @var Model&Searchable&ElasticParams $model */
+                $model = $models[$key];
+
+                if (isset($hit['_score']) && ! empty($hit['_score']) && method_exists($model, 'setElasticsearchScore')) {
```

Comment:
I would prefer that we would call method_exists once for each model type. Not a blocker just possible improvement